### PR TITLE
Build releases for Android.

### PR DESCRIPTION
**Pull request by cjyar**
- Added Android builds to the release workflow.
- Changed how `sdk.dir` is specified, so we don't need to fight with a locally modified file that's committed to the repo.
- Improved the use of the build cache for the CI workflow.

### CODE REVIEW

**Comment by cjyar** [comment]
@clintfred Please re-review.
@coltfred Please check `RELEASING.md`; there are probably some mistakes in there.

There is a release on sonatype snapshots. This can be translated from Maven and used to depend on it:
```
<dependency>
  <groupId>com.ironcorelabs</groupId>
  <artifactId>ironoxide-android</artifactId>
  <version>0.13.1-SNAPSHOT</version>
  <type>aar</type>
</dependency>
```

**Comment by clintfred** [comment]
Opened #117 to remind us to stop depending on cross 0.1.

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+1. Set the version name. The following files need to be updated:
+    - `android/Cargo.toml`
+    - `android/examples/example-instructions.md`
+    - `android/gradle.properties`
+    - `java/Cargo.toml`
+    - `java/tests/version.sbt`
+1. Use `sbt` to release the jar file.
+    1. Make sure you have a sonatype account with write access to com.ironcorelabs. You'll need to type your username/password.
```
Comment (coltfred) [inline RELEASING.md:8]: You don't need this if you have a file in your global sbt configuration, at least for the sbt directions.

**Comment by cjyar** [comment]
> Question: Is it convention to have a `-debug` and `-release` aar available?

I don't know. Maybe we should remove the `-debug`.

File: RELEASING.md (hunk 1 above)
Comment (cjyar) [reply RELEASING.md]: Updated.

**Pull request merged**